### DISCUSSION TITLE
Using "transformer" to add additional properties to markdown

### DISCUSSION THREAD

**Discussion by xarthurx** [nuxt/content#2680] [Q&A] [answered]
Hello, I need the raw content of the markdown file so that I can do customized parsing, but doesn't want to lose what `nuxt-content` has already provided.

I learned that I can use a customized `transformer` to parse markdown:

```ts
import { defineTransformer } from "@nuxt/content/transformers";

export default defineTransformer({
	name: "md-transformer",
	extensions: [".md"],


	parse(_id: string, rawContent: string) {
		return {
			_id,
                        body: rawContent.trim().split('\n').map(line => line.trim()).sort()
		};
	},
});
```
But this will totally change the returned object for `queryContent`.
Is it possible to just add an additional property to the original `Proxy` object returned by `queryContent` instead of totally override it?

**Comment by oneminch** [comment]
Have you explored Nuxt Content hooks? They allow you to hook into states of the parsing process like `content:file:beforeParse` and `content:file:afterParse` to do stuff with your content. It's limited, but it's an officially documented feature ([Hooks](https://content.nuxt.com/recipes/hooks)).

```js
// ~/server/plugins/content.ts

export default defineNitroPlugin((nitroApp) => {
  nitroApp.hooks.hook('content:file:beforeParse', (file) => {
    if (file._id.endsWith('.md')) {
      file.body = file.body.replace(/react/g, 'vue')
    }
  })
})
```

**Reply by xarthurx** [reply, marked as answer]
I tried, but somehow it doesn't work for me.
In the end I come up with this solution:
```ts
export default defineTransformer({
	name: "md-transformer",
	extensions: [".md"],

	parse: async (_id: string, rawContent: string) => {
		// for debugging
		// console.log( "Raw content (first 10 characters):", rawContent.substring(0, 10));

		const parsedContent = await markdownParser.parse(_id, rawContent);
		const { content, data } = matter(rawContent);
    // console.log(content)

		return {
			...parsedContent,
			rawContent: content,
		};
	},
});
```

**Reply by brospars** [reply]
What are `markdownParser` and `matter` ?

**Reply by oneminch** [reply]
I don't think they are functions from Nuxt Content itself. They are probably markdown functions @xarthurx used to demonstrate his approach, or they are from an external library.
